commands/volumes/upload.go

```diff
@@ -56,28 +58,64 @@ func handleUploadVolumeCmd(c client.Client) error {
 		params["Org"] = c.Org
 	}
 
+	var file *os.File
+	var err error
+
+	filename := viper.GetString("path")
+	fi, err := os.Stat(filename)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case fi.IsDir():
+		err = zip.CreateArchiveFromDir(filename)
+		if err != nil {
+			return err
+		}
+		file, err = os.Open(filename + ".tar.bz2")
+		if err != nil {
+			return err
+		}
+	case isRegularFile(filename):
+		err = zip.CreateArchiveFromFile(filename)
+		if err != nil {
+			return err
+		}
+		file, err = os.Open(filename + ".tar.bz2")
+		if err != nil {
+			return err
+		}
+	default: // .bz2 file
+		file, err = os.Open(filename)
+		if err != nil {
+			return err
+		}
+	}
+
+	defer os.Remove(filename + ".tar.bz2")
+
 	subresource := fmt.Sprintf("volume/%s/upload", volume)
 	url := uri.CreateResourceURI("", "environment", envID, subresource, params)
-	form, contentType, err := fileForm(viper.GetString("file"), "volume_tarball")
+	form, contentType, err := fileForm(file, "volume_tarball")
 	if err != nil {
 		return err
 	}
 	_, err = c.Requester.Do(http.MethodPost, url, contentType, form)
```

Comment:
Scoping

Reply:
Can't do that here because I use the `form` and `contentType` values outside the `if` statement.

Reply:
Ack.  This was supposed to be for line 104.  Not sure how that happened